server/api/utils/clients/async_nuclio.py

```diff
@@ -0,0 +1,53 @@
+from mlrun.utils import logger
+import mlrun.utils
+from mlrun.config import config
+import mlrun.common.schemas
+import asyncio
+import urllib.parse
+import http
+import aiohttp
+
+NUCLIO_API_SESSIONS_ENDPOINT = "/api/sessions/"
+NUCLIO_API_GATEWAYS_ENDPOINT = "/api/api_gateways/"
+API_GATEWAY_NAMESPACE_HEADER = "X-Nuclio-Api-Gateway-Namespace"
+NUCLIO_PROJECT_NAME_HEADER = "X-Nuclio-Project-Name"
+
+
+class Client:
+    def __init__(self, auth_info: mlrun.common.schemas.AuthInfo):
+        self._auth = aiohttp.BasicAuth(auth_info.username, auth_info.session)
+        self._session = mlrun.utils.AsyncClientWithRetry(
+            raise_for_status=False, logger=logger
+        )
+        self.nuclio_dashboard_url = mlrun.mlconf.nuclio_dashboard_url
+
+    async def send_request_to_api(self, method, url, path="/", body=None, headers=None):
+        return await self._session.request(
+            method=method,
+            url=urllib.parse.urljoin(url, path),
+            json=body,
+            headers=headers,
+            verify_ssl=False,
+            auth=self._auth,
+        )
+
+    async def list_api_gateways(self, namespace=None, project_name=None):
+        headers = {}
+        if namespace:
+            headers[API_GATEWAY_NAMESPACE_HEADER] = namespace
+
+        if project_name:
+            headers[NUCLIO_PROJECT_NAME_HEADER] = project_name
+
+        response = await self.send_request_to_api(
+            method="GET",
+            url=self.nuclio_dashboard_url,
+            path=NUCLIO_API_GATEWAYS_ENDPOINT,
+            headers=headers,
+        )
+        if response.ok:
+            return await response.json()
+        else:
+            raise mlrun.errors.MLRunInternalServerError(
+                f"Unable to create nuclio api session. Reason: {response.reason}",
+            )
```

Comment:
why not returning the error code / description returned by nuclio to the client?

Reply:
reworked on the same way it was done in iguazio.py